build.sh

```diff
@@ -1,37 +1,38 @@
 #!/bin/bash -e
 # This script is used to build, test and squash the OpenShift Docker images.
 #
-# Name of resulting image will be: 'NAMESPACE/BASE_IMAGE_NAME-VERSION-OS'.
+# Resulting image will be tagged: 'name:version' and 'name:latest'. Name and version
+#                                  are values of labels from resulted image
 #
-# BASE_IMAGE_NAME - Usually name of the main component within container.
-# OS - Specifies distribution - "rhel7" or "centos7"
+# OS - Specifies distribution - "rhel7", "centos7" or "fedora"
 # VERSION - Specifies the image version - (must match with subdirectory in repo)
 # TEST_MODE - If set, build a candidate image and test it
 # TAG_ON_SUCCESS - If set, tested image will be re-tagged as a non-candidate
 #       image, if the tests pass.
 # VERSIONS - Must be set to a list with possible versions (subdirectories)
-# OPENSHIFT_NAMESPACES - Which of available versions (subdirectories) should be
-#       put into openshift/ namespace.
 
 OS=${1-$OS}
 VERSION=${2-$VERSION}
 
 DOCKERFILE_PATH=""
 
-test -z "$BASE_IMAGE_NAME" && {
-  BASE_DIR_NAME=$(echo $(basename `pwd`) | sed -e 's/-[0-9]*$//g')
-  BASE_IMAGE_NAME="${BASE_DIR_NAME#s2i-}"
-}
+BASE_DIR_NAME=$(echo $(basename `pwd`) | sed -e 's/-[0-9]*$//g')
 
 # Perform docker build but append the LABEL with GIT commit id at the end
 function docker_build_with_version {
   local dockerfile="$1"
+  echo "-> Version ${dir}: building image from '${dockerfile}' ..."
+
   git_version=$(git rev-parse --short HEAD)
   BUILD_OPTIONS+=" --label io.openshift.builder-version=\"${git_version}\""
   if [[ "${UPDATE_BASE}" == "1" ]]; then
     BUILD_OPTIONS+=" --pull=true"
   fi
+
+  IMAGE_NAME="$BASE_DIR_NAME-$(cat /dev/urandom | tr -dc 'a-z0-9' | fold -w 4 | head -n 1
```

Comment:
It was used to allow to run `make test` in two separate copies of image directory in parallel.